testcases/elichika_tests/canonicalizer/Return_test.py

```diff
@@ -0,0 +1,265 @@
+import unittest
+import ast, gast
+from chainer_compiler.elichika.parser import canonicalizer
+from chainer_compiler.elichika.parser import utils
+from chainer_compiler.elichika.testtools import compare_ast, assert_semantically_equals
+
+
+class Return(unittest.TestCase):
+    def setUp(self):
+        self.canonicalizer = canonicalizer.Canonicalizer(use_illegal_identifier=False)
+
+    def test_return(self):
+        orig_code = utils.clip_head("""
+        def func(a, b):
+            for i in range(a):
+                if i == b:
+                    return i
+            return 0
+
+        value = 0
+        for a in range(10):
+            for b in range(10):
+                value += func(a, b)
+        """)
+        target_code = utils.clip_head("""
+        def func(a, b):
+            returned_value = None
+            returned_1 = False
```

Comment:
I think we don't need multiple `returned_xx` in a function. Could you modify the code for simplicity?

Reply:
The boolean `returned_1` flag is used to break a loop or when return statement is hit or to skip subsequent statements. The value to be returned is stored in `returned_value`, since different return statements might try to return different values.

I can try combining both the flag and the value variable, by inferring the value of boolean flag based on any change in `returned_value` variable, but it can cause ambiguity when there is a return statement with no return value (i.e. it doesn't change the value of the `returned_value` variable).

Is there an alternative way you have in mind that can work with just a single `returned_xx` variable?

Reply:
Sorry, I'm OK with having both `returned_flag` and `returned_value_key`. I was wondering if we can combine `returned_0` and `returned_1` here.

Reply:
Yeah, I guess that can be done. Actually, maybe this can be done for countinued_xx and breaked_xx flags as well.

Reply:
Alright. Would make the change and see if any edge case arises.

Reply:
Yeah, I was also wondering if we can reduce the number of breaked flags a bit. Thanks for fixing both of them!